pkg/redirectpolicy/redirectpolicy.go

```diff
@@ -0,0 +1,180 @@
+//  Copyright 2020 Authors of Cilium
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package redirectpolicy
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/cilium/cilium/pkg/k8s"
+	v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	k8sUtils "github.com/cilium/cilium/pkg/k8s/utils"
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/policy/api"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// LRPConfig is the internal representation of Cilium Local Redirect Policy.
+type LRPConfig struct {
+	name             string
+	namespace        string
+	frontend         *loadbalancer.L3n4Addr
+	serviceID        *k8s.ServiceID
+	svcPortProto     *api.PortProtocol
+	backendSelector  *api.EndpointSelector
+	backendPortProto *api.PortProtocol
+	backends         []loadbalancer.L3n4Addr
+}
+
+// Parse parses the specified cilium redirect local policy spec, and returns
+// a sanitized LRPConfig.
+func Parse(clrp *v2.CiliumLocalRedirectPolicy, sanitize bool) (LRPConfig, error) {
+	name := clrp.ObjectMeta.Name
+	if name == "" {
+		return LRPConfig{},
+			fmt.Errorf("CiliumLocalRedirectPolicy must have a name")
+	}
+
+	namespace := k8sUtils.ExtractNamespace(&clrp.ObjectMeta)
+	// if namespace == "" {
+	// TODO CCLRP
+	// 	return cclrp.Parse()
+	// }
```

Comment:
nit: Usually we would avoid commented code like this and either just file an issue for tracking the required functionality, or if it's also useful to have a comment line in the code, add a comment in the style:

`// TODO: GH-XXXX: Add support for Clusterwide LRP`